clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -142,9 +139,7 @@ private NetworkClient(MetadataUpdater metadataUpdater,
         this.correlation = 0;
         this.randOffset = new Random();
         this.requestTimeoutMs = requestTimeoutMs;
-        this.nodesEverSeen = new ArrayList<>();
-        this.nodesEverSeenById = new HashMap<>();
-        
+        this.bootstrapNodes = this.metadataUpdater.fetchNodes();
```

Comment:
Passing the bootstrap cluster to Metadata sounds good. That's better than passing an additional constructor parameter called `bootstrapNodes`.